plugins/autodim.koplugin/main.lua

```diff
@@ -0,0 +1,250 @@
+                            if not spin then return end
+                            self.autodim_starttime_m = spin.value
+                            G_reader_settings:saveSetting("autodim_starttime_minutes", spin.value)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                        extra_text = _("Disable"),
+                        extra_callback = function()
+                            self.autodim_starttime_m = -1
+                            G_reader_settings:saveSetting("autodim_starttime_minutes", -1)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                    }
+                    UIManager:show(idle_dialog)
+                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                end,
+                keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    return T(_("Dimmer duration: %1 seconds"), self.autodim_duration_s)
+                end,
+                enabled_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local dimmer_dialog = SpinWidget:new{
+                        title_text = _("Automatic dimmer duration"),
+                        info_text = _("Time is in seconds."),
+                        value = self.autodim_duration_s,
+                        default_value = DEFAULT_AUTODIM_DURATION_S,
+                        value_min = 0,
+                        value_max = 300,
+                        value_step = 1,
+                        value_hold_step = 10,
+                        callback = function(spin)
+                            if not spin then return end
+                            self.autodim_duration_s = spin.value
+                            G_reader_settings:saveSetting("autodim_duration_seconds", spin.value)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                    }
+                    UIManager:show(dimmer_dialog)
+                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                end,
+                keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    return T(_("Dimming percentage: %1%"), self.autodim_endpercentage)
+                end,
+                enabled_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local percentage_dialog = SpinWidget:new{
+                        title_text = _("Percentage"),
```

Comment:
And this here should say what kind of percentage.

```suggestion
                        title_text = _("Dimming percentage"),
```